### PR TITLE
bump aegir from 38.1.8 to 39.0.8

### PR DESCRIPTION
**Pull request by RonaldZielaznicki**
This PR is the culmination of #221 in an effort to upgrade js-stores to use the latest version of aegir.

Dependabot was unable to update to the latest version of aegir due to linting issues. These are similar to the issues that helia ran into a few weeks ago (see: https://github.com/ipfs/helia/pull/111).

This PR namely bumps aegir's version and fixes the resulting linting errors that followed.

### CODE REVIEW

File: packages/datastore-core/src/shard.ts [hunk 1]
```diff
@@ -108,8 +108,6 @@ export function parseShardFun (str: string): Shard {
 
 export const readShardFun = async (path: string | Uint8Array, store: Datastore): Promise<Shard> => {
   const key = new Key(path).child(new Key(SHARDING_FN))
-  // @ts-expect-error
-  const get = typeof store.getRaw === 'function' ? store.getRaw.bind(store) : store.get.bind(store)
```
Comment (RonaldZielaznicki) [inline packages/datastore-core/src/shard.ts:112]: This looks like legacy code that kept getting passed forward. But, as far as I can tell, no store implements `getRaw` nor is one described in the `Store` interface.

File: packages/datastore-core/test/sharding.spec.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ describe('ShardingDatastore', () => {
 
   it('open - empty', () => {
     const ms = new MemoryDatastore()
-    // @ts-expect-error
+    // @ts-expect-error No shard is given on purpose to generate an error.
```
Comment (RonaldZielaznicki) [inline packages/datastore-core/test/sharding.spec.ts:31]: Added a description here.

File: packages/datastore-fs/test/index.spec.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ describe('FsDatastore', () => {
     await fstore.open()
     await ShardingDatastore.create(fstore, new shard.NextToLast(2))
 
-    const file = await fs.readFile(path.join(dir, shard.SHARDING_FN + '.data'))
+    const file = await fs.readFile(path.join(dir, `${shard.SHARDING_FN}.data`))
```
Comment (RonaldZielaznicki) [inline packages/datastore-fs/test/index.spec.ts:118]: I'm not sure why, but the linter was NOT happy with these two strings concatenating with a `+`.

It was happy with a template string though.

File: packages/datastore-s3/test/utils/s3-mock.ts [hunk 4]
```diff
@@ -30,9 +30,9 @@ export const s3Reject = <T> (err: T): any => {
  */
 export function s3Mock (s3: S3): S3 {
   const mocks: any = {}
-  const storage: Map<string, any> = new Map()
+  const storage = new Map<string, any>()
 
-  mocks.send = sinon.replace(s3, 'send', (command) => {
+  mocks.send = sinon.replace(s3, 'send', (command: Parameters<S3['send']>[0]) => {
```
Comment (RonaldZielaznicki) [inline packages/datastore-s3/test/utils/s3-mock.ts:35]: The command that should go here, as far as I could tell, isn't exported by the s3 types. So went with a roundabout way to get the right type.

File: packages/interface-blockstore-tests/package.json [hunk 5]
```diff
@@ -134,13 +134,15 @@
...
-    "aegir": "^38.1.0",
     "interface-blockstore": "^5.0.0",
     "it-all": "^3.0.1",
     "it-drain": "^3.0.1",
     "multiformats": "^11.0.2",
     "uint8arrays": "^4.0.2"
   },
+  "devDependencies": {
+    "aegir": "^39.0.8"
+  },
```
Comment (RonaldZielaznicki) [inline packages/interface-blockstore-tests/package.json:145]: Moved aegir to devDependencies.

File: packages/interface-blockstore/src/index.ts [hunk 6]
```diff
@@ -10,10 +10,10 @@ export interface Pair {
...
 }
 
-export interface Blockstore <HasOptionsExtension = {},
-PutOptionsExtension = {}, PutManyOptionsExtension = {},
-GetOptionsExtension = {}, GetManyOptionsExtension = {}, GetAllOptionsExtension = {},
-DeleteOptionsExtension = {}, DeleteManyOptionsExtension = {}> extends Store<CID, Uint8Array, Pair, HasOptionsExtension,
+export interface Blockstore <HasOptionsExtension = unknown,
+PutOptionsExtension = unknown, PutManyOptionsExtension = unknown,
+GetOptionsExtension = unknown, GetManyOptionsExtension = unknown, GetAllOptionsExtension = unknown,
+DeleteOptionsExtension = unknown, DeleteManyOptionsExtension = unknown> extends Store<CID, Uint8Array, Pair, HasOptionsExtension,
```
Comment (RonaldZielaznicki) [inline packages/interface-blockstore/src/index.ts:16]: `{}` wasn't well liked by the linter. Thought maybe `Record<string, unknown` would be better. Eventually settled on unknown to solve the linter's complaints but would love a more specific type if someone knows one that is better.

File: packages/interface-datastore-tests/package.json [hunk 7]
```diff
@@ -134,14 +134,16 @@
...
     "interface-datastore": "^8.0.0",
     "iso-random-stream": "^2.0.0",
     "it-all": "^3.0.1",
     "it-drain": "^3.0.1",
     "it-length": "^3.0.1",
     "uint8arrays": "^4.0.2"
   },
+  "devDependencies": {
+    "aegir": "^39.0.8"
+  },
```
Comment (RonaldZielaznicki) [inline packages/interface-datastore-tests/package.json:146]: moved aegir over as a devDependency

File: package.json [hunk 8]
```diff
@@ -34,8 +34,8 @@
     "dep-check": "aegir run dep-check",
     "release": "npm run docs:no-publish && aegir run release && npm run docs"
   },
-  "dependencies": {
-    "aegir": "^38.1.0"
+  "devDependencies": {
+    "aegir": "^39.0.8"
```
Comment (RonaldZielaznicki) [inline package.json:38]: moved aegir over as a devDependency

**Comment by achingbrain** [comment]
Ah, thanks so much for picking this up, I'm really sorry I missed it though & the aegir upgrade was done in https://github.com/ipfs/js-stores/pull/225

Thanks for opening this though!

**Pull request closed without merging**